extension/src/runtime.ts

```diff
@@ -11,21 +11,15 @@ export const launchRuntime = (terminals: Map<string, vscode.Terminal>): void =>
     const isWindowsPlatform = /^win/.test(process.platform);
     const mono = (isWindowsPlatform) ? "" : "mono";
     const xcassemblies = `${rootPath}${path.sep}XCR${path.sep}xcassemblies`;
-    const xcrFiles = [];
-
-    fs.readdirSync(xcassemblies).forEach(file => {
-        if (file.endsWith(xcrExtension)) {
-            xcrFiles.push(file);
-        }
-    });
-
-    xcrFiles.forEach(file => {
-        if (terminals[file] === undefined) {
-            terminals[file] = vscode.window.createTerminal(file);
-        }
-        terminals[file].show();
-        terminals[file].sendText(`${mono} ${xcruntimePath} ${xcassemblies}${path.sep}${file}`);
-    });
+    const xcrFiles = fs.readdirSync(xcassemblies)
+        .filter(file => file.endsWith(xcrExtension))
+        .map(file => {
```

Comment:
Strange use of map (your lambda returns nothing)
Use map to convert a list of files into something else (terminal list...) if needed
Or perhaps just use filter and foreach